pyserini/trectools/_base.py

```diff
@@ -110,7 +110,7 @@ def read_run(self, filepath: str, resort: bool = False) -> None:
         self.run_data = pd.read_csv(filepath, sep='\s+', names=TrecRun.columns, dtype={'docid': 'str'})
         if resort:
             self.run_data.sort_values(["topic", "score"], inplace=True, ascending=[True, False])
-            self.run_data["rank"] = self.run_data.groupby("topic")[["docid","score"]].rank(ascending=False,method='first')
+            self.run_data["rank"] = self.run_data.groupby("topic")["score"].rank(ascending=False,method='first')
```

Comment:
Why wouldn't we have the same number of docid/score? Is there a deeper bug elsewhere?

Reply:
We do have the same number of docid and score, just .rank() should return one column instead of two columns. In version 1.1.4, [["docid","score"]].rank() returns the rank of the sorted score column, but in newer version it also returns the docid string column. Thus, with pandas==1.3.5, assigning two columns into one caused the ValueError: Columns must be same length as key

Reply:
Ah, gotcha, thanks for the explanation!

Reply:
Thanks a lot! I'm still running a full regression test on clprf experiments with new packages version. Will merge it after all tests are done.